list.go

```diff
@@ -0,0 +1,91 @@
+package pack
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"reflect"
+)
+
+type List struct {
+	T     Type
+	Elems []Value
+}
+
+func NewList(vs ...interface{}) List {
```

Comment:
This function should check that all of the elements are of the same type. If not, it should panic. This will help prevent accidentally constructing an invalid list.